src/main/java/org/jabref/model/groups/WordKeywordGroup.java

```diff
@@ -113,6 +118,12 @@ public boolean contains(BibEntry entry) {
 
     private Set<String> getFieldContentAsWords(BibEntry entry) {
         if (onlySplitWordsAtSeparator) {
+            if (BibEntry.TYPE_HEADER.equals(searchField)) {
+                Optional<EntryType> entryType = EntryTypes.getType(entry.getType(), BibDatabaseMode.BIBLATEX);
+                if (entryType.isPresent()) {
+                    return searchWords.stream().filter(sw -> entryType.get().getName().equals(sw)).collect(Collectors.toSet());
```

Comment:
What's the reason for filtering the `searchWords` list? I think you can safely return the value of `entryType` (as a 1-item list).

Reply:
<code>searchWords</code> is a 1-item set. Filter method lets me compare the content of this set with the entry type. Example, searchWord can be 'Book' but the entry type can be 'Article' and hence, this entry won't fall under Book subgroup. Other approach that I could think of -

<code>if (entryType.isPresent() && searchWords.iterator().next().equals(entryType.get().getName())) {</code>
	<code>return searchWords;</code>
<code>}</code>

Please suggest.

Reply:
Your solution sounds reasoable, I would let it as is.